unittests/Parse/LexerTests.cpp

```diff
@@ -731,29 +763,6 @@ TEST_F(LexerTest, DiagnoseEmbeddedNul) {
                              "1, 4: nul character embedded in middle of file"));
 }
 
-TEST_F(LexerTest, DiagnoseEmbeddedNulOffset) {
-  const char Source[] = " \0 \0 aaa \0 \0 bbb";
-  size_t SourceLen = sizeof(Source) - 1;
-
-  LangOptions LangOpts;
-  SourceManager SourceMgr;
-  unsigned BufferID = SourceMgr.addMemBufferCopy(StringRef(Source, SourceLen));
-
-  StringCaptureDiagnosticConsumer DiagConsumer;
-  DiagnosticEngine Diags(SourceMgr);
-  Diags.addConsumer(DiagConsumer);
-
-  Lexer L(LangOpts, SourceMgr, BufferID, &Diags,
-          LexerMode::Swift, HashbangMode::Disallowed,
-          CommentRetentionMode::None,
-          /*Offset=*/5, /*EndOffset=*/SourceLen);
```

Comment:
Yes, because offset is 5, we shouldn’t receive any diagnostics for the first two null bytes in `Source`. What I was saying in my last comment was basically: Let’s also assert that that’s the case.